src/frida/repl.py

```diff
@@ -423,6 +431,57 @@ def _create_repl_script(self):
                 with codecs.open(self._user_script, 'rb', 'utf-8') as f:
                     user_script = f.read().rstrip("\r\n") + "\n\n// Frida REPL script:\n"
 
+            if self._codeshare_uri is not None:
+                trust_store = self._get_or_create_truststore()
+                project_url = 'https://codeshare.frida.re/api/project/{}/'.format(self._codeshare_uri)
+                response_json = None
+                try:
+                    response = urlopen(project_url)
+                    response_content = response.read()
+                    response_json = json.loads(response_content)
+                except Exception as e:
+                    self._print("Got an unhandled exception while trying to retrieve {} - {}".format(self._codeshare_uri, e))
+
+                if response_json:
+                    trusted_signature = trust_store.get(self._codeshare_uri, '')
+                    fingerprint = hashlib.sha256(response_json['source'].encode('utf-8')).hexdigest()
+                    if fingerprint == trusted_signature:
+                        user_script = response_json['source']
+                    else:
+                        self._print("""Hello! This is the first time you're running this particular snippit, or the snippit's source code has changed.
```

Comment:
Is it a `snippit` or a `snippet`? I have no idea, you're the native speaker, so definitely your call! :)